packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -166,19 +181,43 @@ export const SimpleFormIterator = (props: SimpleFormIteratorProps) => {
                         </SimpleFormIteratorItem>
                     ))}
                 </ul>
-                {!disabled && !disableAdd && (
-                    <div className={SimpleFormIteratorClasses.add}>
-                        {cloneElement(addButton, {
-                            className: clsx(
-                                'button-add',
-                                `button-add-${source}`
-                            ),
-                            onClick: handleAddButtonClick(
-                                addButton.props.onClick
-                            ),
-                        })}
-                    </div>
-                )}
+                <div className={SimpleFormIteratorClasses.buttons}>
+                    {!disabled && !disableAdd && (
+                        <div className={SimpleFormIteratorClasses.add}>
+                            {cloneElement(addButton, {
+                                className: clsx(
+                                    'button-add',
+                                    `button-add-${source}`
+                                ),
+                                onClick: handleAddButtonClick(
+                                    addButton.props.onClick
+                                ),
+                            })}
+                            {fields.length > 0 && (
```

Comment:
I realize we should not display that button if `disableRemove` is set.

Reply:
resolve commit : [d8f3dc5](https://github.com/marmelab/react-admin/pull/8302/commits/d8f3dc54cc96e07dc501a394cf32488235fe9a86)